src/Altinn.App.Api/Controllers/OrganisationSearchController.cs

```diff
@@ -0,0 +1,53 @@
+using System.Net.Mime;
+using Altinn.App.Api.Infrastructure.Filters;
+using Altinn.App.Api.Models;
+using Altinn.App.Core.Internal.Registers;
+using Altinn.Platform.Register.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Altinn.App.Api.Controllers;
+
+/// <summary>
+/// This controller class provides Enhetsregisteret (ER) organisation search functionality.
+/// </summary>
+[AutoValidateAntiforgeryTokenIfAuthCookie]
+[ApiController]
+[Produces(MediaTypeNames.Application.Json)]
+[Consumes(MediaTypeNames.Application.Json)]
+[ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
+[Route("{org}/{app}/api/v1/organisations")]
+public class OrganisationSearchController : ControllerBase
+{
+    private readonly IOrganizationClient _organisationClient;
+    private readonly ILogger<OrganisationSearchController> _logger;
+
+    /// <summary>
+    /// Initialize a new instance of <see cref="OrganisationSearchController"/> with the given services.
+    /// </summary>
+    /// <param name="organisationClient">A client for searching for an organisation in ER.</param>
+    /// <param name="logger">A logger for logging.</param>
+    public OrganisationSearchController(
+        IOrganizationClient organisationClient,
+        ILogger<OrganisationSearchController> logger
+    )
+    {
+        _organisationClient = organisationClient;
+        _logger = logger;
+    }
+
+    /// <summary>
+    /// Allows searching for an organisation in ER
+    /// </summary>
+    /// <param name="orgNr">Payload that contains params for executing a search for an organisation.</param>
+    /// <returns>A <see cref="OrganisationSearchResponse"/> object.</returns>
+    [HttpGet]
+    [Route("{orgNr}")]
+    [ProducesResponseType(typeof(OrganisationSearchResponse), StatusCodes.Status200OK)]
+    public async Task<ActionResult<OrganisationSearchResponse>> SearchForOrganisation([FromRoute] string orgNr)
```

Comment:
I'm 100% comfortable with this being a GET with orgNr as route param, but I still want to ask:
Consider following same pattern as the one for Person, just to keep these two consistent with each other?
Meaning POST with a JSON request object that contains the parameters. Then the frontend components can be more similar as well.

The reason it was done that way for person was to avoid having the SSN in the URL, so I get that the same reason is not valid here.

Reply:
I see what you mean, and I did consider it. But I think I'd rather adhere to REST guidelines as a base case, and make exceptions when it makes sense, like for the person lookup. For the frontend, this needs to separate queries anyways, so there won't be much difference either way.

Feel free to disagree 😅